File: solutions/typescript/0007.ts
function reverse(x: number): number {
    const MAX_INT = Math.pow(2, 31) - 1;
    const MIN_INT = -Math.pow(2, 31);
    
    const isNegative = x < 0;
    const reversed = parseInt(Math.abs(x).toString().split("").reverse().join(""));
    const result = isNegative ? -reversed : reversed;
    
    // 检查结果是否在有效范围内
    if (result > MAX_INT || result < MIN_INT) {
        return 0;
    }
    
    return result;
}

File: solutions/python/0007.py
class Solution:
    def reverse(self, x: int) -> int:
        MAX_INT = 2**31 - 1
        MIN_INT = -2**31
        is_negative = x < 0
        reversed = int(str(abs(x))[::-1])
        result = -reversed if is_negative else reversed
        return result if MIN_INT <= result <= MAX_INT else 0